helm/charts/determined/templates/genai/genai-deployment.yaml

```diff
@@ -0,0 +1,77 @@
+{{- if not .Values.genai.version }}
+{{ else }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: genai-deployment-{{ .Release.Name }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: genai-{{ .Release.Name }}
+    release: {{ .Release.Name }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: genai-{{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: genai-{{ .Release.Name }}
+        determined-system: master
+    spec:
+      priorityClassName: determined-system-priority
+      serviceAccount: genai-{{ .Release.Name }}
+      containers:
+      - name: genai-{{ .Release.Name }}
+        {{ $image := "environments-dev" }}
+        {{ $tag := (required "A valid .Values.genai.version entry required!" .Values.genai.version) }}
+        {{- /* genai.version is used for CI to override the appVersion. */ -}}
+        image: {{ .Values.imageRegistry }}/{{ $image }}:lore-backend-image-{{ $tag }}
+        imagePullPolicy: "Always"
+        env:
+          - name: DET_MASTER
+            value: determined-master-service-{{ .Release.Name }}:{{ .Values.masterPort }}
+          - name: DB_NAME
+            value: lore
+          - name: DB_USER
+            value: postgres
+          - name: DB_PASSWORD
+            value: postgres
+          - name: DB_PORT
+            value: "5432"
+          - name: DB_HOST
+            value: determined-db-service-{{ .Release.Name }} 
```

Comment:
these may not be default, so this should to the same thing as this master config section: https://github.com/determined-ai/determined/blob/main/helm/charts/determined/templates/master-config.yaml#L47-L56